File: php/testcode/codeinj_016.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_codeinj_usort_closure
function codeinj016(BenchmarkRequest $req): BenchmarkResponse {
    $items = explode(',', $req->param('items'));
    usort($items, function($a, $b) { // vuln-code-snippet safe-line php_codeinj_usort_closure
        return strcmp($a, $b);
    });
    return BenchmarkResponse::json($items);
}
// vuln-code-snippet end php_codeinj_usort_closure

File: ruby/testcode/codeinj_016.rb
require_relative 'shared'
require 'json'

# vuln-code-snippet start ruby_codeinj_json_schema
def parse_json_safe(req)
  data = JSON.parse(req.body_str) # vuln-code-snippet safe-line ruby_codeinj_json_schema
  BenchmarkResponse.json(data)
end
# vuln-code-snippet end ruby_codeinj_json_schema
